src/main/java/net/snowflake/client/core/SessionUtil.java

```diff
@@ -584,9 +588,38 @@ private static SFLoginOutput newSession(
 
       setServiceNameHeader(loginInput, postRequest);
 
-      String theString =
-          HttpUtil.executeGeneralRequest(
-              postRequest, loginInput.getLoginTimeout(), loginInput.getHttpClientSettingsKey());
+      String theString = null;
+
+      while (true) {
+        try {
+          theString =
+              HttpUtil.executeGeneralRequest(
+                  postRequest,
+                  loginInput.getLoginTimeout(),
+                  loginInput.getAuthTimeout(),
+                  retryCount,
+                  loginInput.getHttpClientSettingsKey());
+        } catch (SnowflakeSQLException ex) {
+          if (ex.getMessage().contains("Authenticator Request Timeout")) {
```

Comment:
Is there an error code or something that comes with the exception that could be checked instead of checking the error message? I worry that sometimes the text of exception messages changes with library updates and such, but usually the code stays consistent.

Reply:
I updated this to use an error code instead of a message!